packages/api/src/errors.js

```diff
@@ -77,3 +77,63 @@ export class InvalidCidError extends Error {
   }
 }
 InvalidCidError.CODE = 'ERROR_INVALID_CID'
+
+export class PinningServiceApiError {
+  /**
+   *
+   * @param {string} reason
+   * @param {string} [details]
+   * @param {number} [status]
+   */
+  constructor (reason = 'PSA_ERROR', details, status = 400) {
+    this.reason = reason
+    this.details = details
+    this.status = status
+  }
+}
+
+export class PSAErrorInvalidData extends PinningServiceApiError {
+  /**
+   * @param {string} details
+   */
+  constructor (details = 'Some data is invalid. Please consult the spec.') {
+    super(PSAErrorInvalidData.REASON)
+    this.details = details
+  }
+}
+PSAErrorInvalidData.REASON = 'PSA_INVALID_DATA'
+
+export class PSAErrorRequiredData extends PinningServiceApiError {
+  /**
+   * @param {string} details
+   */
+  constructor (details = 'Missing required data. Please consult the spec.') {
+    super(PSAErrorRequiredData.REASON)
+    this.details = details
+  }
+}
+PSAErrorRequiredData.REASON = 'PSA_REQUIRED_DATA'
+
+export class PSAErrorDBNotFound extends PinningServiceApiError {
+  /**
+   * @param {string} details
+   */
+  constructor (details = 'Requested data was not found.') {
+    super(PSAErrorDBNotFound.REASON)
+    this.details = details
+    this.status = 404
+  }
+}
+PSAErrorDBNotFound.REASON = 'PSA_DB_NOT_FOUND'
```

Comment:
Maybe this could be named differently. I read it as meaning "database not found". Is it `PSAErrorResourceNotFound` ?

Reply:
Thanks! I'm not good with naming things, wasn't happy with this one either 😅